src/base/common.zig

```diff
@@ -10,11 +10,21 @@ pub const MessageHeader = struct {
     rsv1: bool = false,
     rsv2: bool = false,
     rsv3: bool = false,
-    opcode: u4,
+    opcode: Opcode,
     length: u64,
     mask: ?u32 = null,
 };
 
+pub const Opcode = enum(u4) {
```

Comment:
This should probably explicitly set the enum values, I don't trust zig to not mangle this during optimizations.

Reply:
Ok, updated in 514019d. I would be surprised if optimizations did that, though, since it seems like it would break a lot of programs.